transformer_engine/jax/csrc/modules.cpp

```diff
@@ -376,7 +376,7 @@ void LayerNormBackwardImpl(size_t n, size_t hidden, bool zero_centered_gamma, fl
 
     TensorWrapper dummy_workspace_tensor, dummy_barrier_tensor;
     TensorWrapper dummy_dgamma_part_tensor, dummy_dbeta_part_tensor;
-    auto num_sm = cudaDevicePropertiesManager::Instance().GetMultiProcessorCount();
+    auto num_sm = cudaDevicePropertiesManager::Instance().GetMultiProcessorCount() - sm_margin;
```

Comment:
If there is no way to control this parameter from TE-JAX Python API, it might be better to just use `std::getenv` on the C++ side and entirely avoid the PyBind + JAX descriptor mechanism. What do you think?

Reply:
From the @ksivaman's feedback, frequently accessing environment variables can negatively impact performance. Placing on the python side will only ask the environment variables once which should have the least impact on performance.

Reply:
That's a good point.

That said, the custom ops workspace changes in #532 introduce new C++ API for querying workspace sizes in the JAX primitive abstract, and that should only be evaluated once. It could be possible to put the `transformer_engine::getenv` in there to simplify the code without performance impact.

I'm approving the PR regardless so we can merge and then address this later if need be. Thanks!

Reply:
Cool, thanks!